cognite/client/_api/user_profiles.py

```diff
@@ -0,0 +1,154 @@
+        return self._list(
+            "GET",
+            list_cls=UserProfileList,
+            resource_cls=UserProfile,
+            limit=limit,
+            initial_cursor=initial_cursor,
+        )
+
+    def retrieve(self, user_identifier: str) -> UserProfile | None:
+        """`Retrieve a single user profile by user identifier. <https://developer.cognite.com/api#tag/User-profiles/paths/~1profiles~1byids/post>`_
+
+        Retrieves one user profile as indexed by the user identifier in the same CDF project.
+
+        Args:
+            user_identifier (str): The user identifier of the user profile to retrieve.
+
+        Returns:
+            UserProfile | None: The requested user profile or None if it doesn't exist.
+
+        Examples:
+
+            Get user profile by user identifier:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.iam.user_profiles.retrieve("foo")
+        """
+        identifier = UserIdentifierSequence.load(user_identifier).as_singleton()
+        return self._retrieve_multiple(
+            list_cls=UserProfileList,
+            resource_cls=UserProfile,
+            identifiers=identifier,
+        )
+
+    def retrieve_multiple(self, user_identifiers: Sequence[str]) -> UserProfileList:
+        """`Retrieve multiple user profiles by user identifier. <https://developer.cognite.com/api#tag/User-profiles/paths/~1profiles~1byids/post>`_
+
+        Retrieves one or more user profiles indexed by the user identifier in the same CDF project.
+
+        Args:
+            user_identifiers (Sequence[str]): The list of user identifiers to retrieve.
+
+        Returns:
+            UserProfileList: The requested user profiles.
+
+        Raises:
+            CogniteNotFoundError: One or more user identifiers does not exist.
+
+        Examples:
+
+            Get user profiles by user identifier:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.iam.user_profiles.retrieve_multiple(["foo", "bar"])
+        """
+        if isinstance(user_identifiers, str):
+            user_identifiers = [user_identifiers]
+
+        identifiers = UserIdentifierSequence.load(user_identifiers)
+        unsorted_profiles = cast(
+            UserProfileList,
+            self._retrieve_multiple(
+                list_cls=UserProfileList,
+                resource_cls=UserProfile,
+                identifiers=identifiers,
+            ),
+        )
+        # TODO: The API does not guarantee any ordering (against style guidelines, no timeline for fix)
+        #       so we sort manually for now:
+        return UserProfileList(
+            cast(List[UserProfile], [unsorted_profiles.get(user) for user in user_identifiers]),
+            cognite_client=unsorted_profiles._cognite_client,
+        )
+
+    def search(self, name: str, limit: int = 100) -> UserProfileList:
```

Comment:
Why do you change the default limit? Suggest keeping it in line with all other listing type endpoints. 
```suggestion
    def search(self, name: str, limit: int = LIST_LIMIT_DEFAULT) -> UserProfileList:
```

Reply:
See other PR